app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt

```diff
@@ -295,12 +302,44 @@ class UrlInputFragment :
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = if (FeatureFlags.isMvp) {
-        inflater.inflate(R.layout.fragment_urlinput2, container, false)
-    } else {
-        inflater.inflate(R.layout.fragment_urlinput, container, false)
+    ): View? {
+        val view = if (FeatureFlags.isMvp) {
+            inflater.inflate(R.layout.fragment_urlinput2, container, false)
+        } else {
+            inflater.inflate(R.layout.fragment_urlinput, container, false)
+        }
+
+        val topSites = view.findViewById<ComposeView>(R.id.topSites)
+        topSites.setContent {
```

Comment:
Does this even work with the old layout where this view doesn't exist? :) But maybe just get rid of the branching based on the flag here.

Reply:
This would crash with the old layout. But as we do not need this anymore on main....